load-generator-source/src/main/java/org/springframework/cloud/stream/module/loadgenerator/LoadGeneratorSource.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.loadgenerator;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.integration.endpoint.AbstractEndpoint;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * A source that sends a set amount of empty byte array messages to verify the speed
+ * of the infrastructure.
+ *
+ * @author Glenn Renfro
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({LoadGeneratorSourceProperties.class})
+public class LoadGeneratorSource extends AbstractEndpoint {
+
+	private static final Logger logger = LoggerFactory.getLogger(LoadGeneratorSource.class);
+
+	@Autowired
+	private volatile LoadGeneratorSourceProperties config;
```

Comment:
After some discussion with @markfisher I've decided to remove the `volatile` since the same thread that performs the auto wiring will also invoke `doStart()`.